### PR TITLE
Craete our own enumerables

### PR DESCRIPTION
**Pull request by Kandy-Peter**
- [ ] Create a class `MyList` that has an instance variable `@list`.

- [ ] In MyList implement a method #each that yields successive members of @list and uses the My[Enumerable](https://ruby-doc.org/core-3.0.0/Enumerable.html) module.

- [ ] Create a module MyEnumerable that implements the following methods (they should have the same funcionality as methods in Enumerable):
      - `#all`?
      - `#any`?
      - `#filter`

### CODE REVIEW

File: src/my_list.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+# Test #any?
+list.any? { |e| e == 2 }
+# => true
+
+list.any? { |e| e == 5 }
+# => false
+
+# Test #filter
+list.filter(&:even)
+# => [2, 4]
```
Comment (Emmyn5600) [inline src/my_list.rb]: - [ ] Let us fix the following issues by adding the `?` sign after the filter and even methods.

image

image

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi @Kandy-Peter and @Lichapa 

Good job so far! I know you have worked hard on making this milestone. There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there

## Required Changes :recycle:

- [ ] Check the comment under the review


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!:clap::clap::clap:
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/my_list.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+# Test #any?
+list.any? { |e| e == 2 }
+# => true
+
+list.any? { |e| e == 5 }
+# => false
+
+# Test #filter
+list.filter(&:even?)
```
Comment (er-contreras) [inline src/my_list.rb]: - [ ] Good Job but do not forget to add the quote mark right after the filter method as the last CR suggests.😉

**Review by er-contreras — CHANGES_REQUESTED** [review]
Hi @Lichapa, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

almost there

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by SajjadAhmad14 — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

### Hi team, 
## Congratulations 🎉 Thank you for implementing all the requirements 🙏
### Please go ahead and merge the branch!

## Happy Coding 👏👏👏
#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**